Makefile

```diff
@@ -8,268 +27,510 @@ usage:
+	@echo "       allboards: Compiles Tock for all supported boards"
 	@echo "        allcheck: Checks, but does not compile, Tock for all supported boards"
 	@echo "          alldoc: Builds Tock documentation for all boards"
-	@echo "           audit: Audit Cargo dependencies for all kernel sources"
-	@echo "              ci: Run all continuous integration tests"
 	@echo "           clean: Clean all builds"
-	@echo "          clippy: Runs the clippy code linter with Tock's default arguments"
-	@echo " emulation-check: Run the emulation tests for supported boards"
-	@echo " emulation-setup: Setup QEMU for the emulation tests"
 	@echo "          format: Runs the rustfmt tool on all kernel sources"
-	@echo "    format-check: Checks if the rustfmt tool would require changes, but doesn't make them"
 	@echo "            list: Lists available boards"
 	@echo
+	@echo "The make system also drives all continuous integration and testing:"
+	@echo "         $$(tput bold)prepush$$(tput sgr0): Fast checks to run before pushing changes upstream"
+	@echo "          ci-all: Run all continuous integration tests (possibly slow!)"
+	@echo "         ci-help: More information on Tock CI and testing"
+	@echo
 	@echo "$$(tput bold)Happy Hacking!$$(tput sgr0)"
 
+##
+## End: usage.
+##
+################################################################################
+##
+## Utility functions
+##
+
+define banner
+	@printf "\n"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@string="$(1)" && printf "$$(tput bold)* %-$$((76))s *\n" "$$string"
+	@printf "$$(tput bold)********************************************************************************$$(tput sgr0)\n"
+	@printf "\n"
+endef
+
+# Four arguments:
+#  1) Command to check if already installed
+#  2) String that explains what will be executed if install runs
+#  3) Make function that does the work
+#  4) Guard variable that is defined if job is to run
+define ci_setup_helper
+	@# The line continuation adds a leading space, remove
+	$(eval explanation := $(strip $(2)))
+	$(eval build_function := $(strip $(3)))
+	$(eval guard_variable := $(strip $(4)))
+	@# First, if the dependency is installed, we can bail early
+	$(if $(shell bash -c '$(1)'),$(eval $(guard_variable) := true),
+	@# If running in CI context always yes
+	$(if $(CI),$(eval do_install := yes_CI),
+	@# If running nosetup always no
+	$(if $(TOCK_NOSETUP),$(eval do_install := ),
+	@# Otherwise, ask
+	$(info )
+	$(info Tock needs to install something to run this step.)
+	$(info )
+	$(info You have run a (likely CI) rule that requires Tock to run setup commands on your)
+	$(info machine. Tock can do this automatically for you, or you can look at the recipe)
+	$(info for '$(build_function)' and install yourself.)
+	$(info )
+	$(info Continuing will: $(explanation).)
+	$(info )
+	$(info You can use 'make ci-nosetup' to run all CI with no new install requirements.)
```

Comment:
Same issue with install

Reply:
Removed that, and one other one as well.

Reply:
Thanks! Sorry to be a pain, I think it just turns people off running the tests if you tell them we need to install things.